docs/configuration.asciidoc

```diff
@@ -671,16 +671,20 @@ The default value `0` means that no source code will be collected for span libra
 ==== `errorMessageMaxLength`
 
 * *Type:* String
-* *Default:* `2kb`
+* *Default:* `"2kb"`
 * *Env:* `ELASTIC_APM_ERROR_MESSAGE_MAX_LENGTH`
 
+This option is **deprecated** -- use <<long-field-max-length,`longFieldMaxLength`>> instead.
+
 The maximum length allowed for error messages.
 It is expressed in bytes or includes a size suffix such as `2kb`.
 Size suffixes are case-insensitive and include `b`,
 `kb`,
 `mb`,
 and `gb`.
 Messages above this length will be truncated before being sent to the APM Server.
+Note that while the configuration option accepts a number of *bytes*, truncation
```

Comment:
nit: As written, I'm a little uncertain what this extra message is trying to convey about the behavior of `errorMessageMaxLength`, and how the bytes configured will be converted into characters.

Reply:
Given the field is now deprecated, should I just drop that comment to avoid adding confusion?

What is happening is that we accept a config value that is "some number of bytes" (because the suffixes are "b", "kb", etc.). However that is just converted to an integer: `"2kb"` becomes `2048`. Then that `2048` is used as a count of *characters* (not bytes) for translation. It is a little subtle, but almost certainly won't affect anyone -- only if they were having error messages near the limit and the error messages had lots of emoji or chars outside the BMP in them.

Reply:
Ah -- I see, so the string is converted into a byte value, and then that byte value is used as a character length.  I'm not sure whether it's better to say nothing or keep what we have to indicate that's it's not a true limiting by byte size.  Nit withdrawn.